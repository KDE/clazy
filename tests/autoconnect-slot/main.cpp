#include <QtCore/QObject>
#include "ui_widget.h"

class MyObject : public QObject, private Ui::Widget
{
    Q_OBJECT
public:
    bool on_mButton_something(); // No warn

public Q_SLOTS:
    void slot1() const {} // No warn
    void slot2_something() {} // No warn
    void slot3_something_(); // No warn
    void on_mButton_clicked(); // Warn
    void on_button_clicked(); // Warn
    void on_mButton_enabled(); // Warn
    void on_mButton_released() {} // Warn
    void on_mButton(); // No warn
    void on_notButton_clicked(); // No warn
    void on_mDouble_clicked(); // No warn - not a widget

  private:

    double mDouble;
    QPushButton* button;

};

void MyObject::on_mButton_enabled() // OK, already warned
{

}

void test()
{
    MyObject o;
}
