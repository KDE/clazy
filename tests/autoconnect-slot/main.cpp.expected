autoconnect-slot/main.cpp:14:5: warning: Use of autoconnected UI slot: on_mButton_clicked [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:15:5: warning: Use of autoconnected UI slot: on_button_clicked [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:16:5: warning: Use of autoconnected UI slot: on_mButton_enabled [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:17:5: warning: Use of autoconnected UI slot: on_mButton_released [-Wclazy-autoconnect-slot]
